usr/share/rear/restore/BORG/default/30_load_archives.sh

```diff
@@ -0,0 +1,62 @@
+# This file is part of Relax-and-Recover, licensed under the GNU General
+# Public License. Refer to the included COPYING for full text of license.
+#
+# 10_load_archives.sh
+
+LogPrint "Starting Borg restore"
+
+# Do we have Borg binary?
+# Missing Borg binary should not happen as it is copyied by
+# backup/BORG/default/20_start_backup.sh
+has_binary borg
+BugIfError "Could not find Borg binary"
+
+# Query Borg server for repoitory information and store it to archive_cache.
+# This should avoid repeatingly quering Borg server, which could be slow.
+archive_cache=$TMP_DIR/borg_archive
+borg list $BORG_USERNAME@$BORG_HOST:$BORG_REPO > $archive_cache
+StopIfError "Could not list Borg archive"
+
+# Store number of lines in archive_cache file for later use
+archive_cache_lines=$(wc -l $archive_cache | awk '{print $1}')
+
+# This means empty repository
+if [ $archive_cache_lines -eq 0 ]; then
+    Error "Borg repository $BORG_REPO on $BORG_HOST is empty"
+fi
```

Comment:
@ThomasWaldmann  
Let me roughly describe how ReaR restore works.

1. Boot small ISO image that contains some base system (including your configuration at the time backup was running)
2. Start restore with `rear recover`
3. ReaR loads and and creates your file system layout
4  ReaR starts restore of your data (this is stage when Borg comes to game)
5. ReaR does some exit tasks like boot loader installation, etc.

As you can see outputting empty list will you no good ... 
In my opinion result must be failure as you can't correctly restore your OS. Of course after such failure you can reconfigure ReaR recovery system to use different settings (e.g. different Borg repository, different file system layout) and try to run restore again.